cap-primitives/src/rustix/fs/stat_unchecked.rs

```diff
@@ -15,5 +23,26 @@ pub(crate) fn stat_unchecked(
         FollowSymlinks::No => AtFlags::SYMLINK_NOFOLLOW,
     };
 
+    // stax is preferred on Linux because it can return creation times. Linux
+    // kernels prior to 4.11 don't have statx and return ENOSYS. We store the
+    // availability in a global to avoid unnecessary syscalls.
+    #[cfg(all(target_os = "linux", target_env = "gnu"))]
+    static HAS_STATX: AtomicBool = AtomicBool::new(true);
+
+    #[cfg(all(target_os = "linux", target_env = "gnu"))]
+    if HAS_STATX.load(Ordering::Relaxed) {
+        let statx_result = statx(
+            start,
+            path,
+            atflags,
+            StatxFlags::BASIC_STATS | StatxFlags::BTIME,
+        );
+        if let Err(rustix::io::Error::NOSYS) = statx_result {
+            HAS_STATX.store(false, Ordering::Relaxed);
+        } else {
+            return Ok(statx_result.map(MetadataExt::from_rustix_statx)?);
+        }
```

Comment:
As a minor style nit, I prefer to do a single `match` on an `rustix::io::Result`, rather than picking out one error case with an `if let` and handling other error cases with `.map`, like this:
```rust
match statx_result {
  Ok(statx) => return Ok(...),
  Err(rustix::io::Error::NOSYS) => ...
  Err(e) => return Err(e),
}
```

Reply:
I went back and forth on that, but the single match is probably better. Updating.

Reply:
Updated in c7f65a2.